### ISSUE TITLE
Upgrading to 8.0.0-preview2 broke some dependencies

### ISSUE THREAD

**Issue by kooshan** [OData/AspNetCoreOData#30]
```
TypeLoadException: Could not load type 'Microsoft.AspNet.OData.Query.HandleNullPropagationOption' from assembly 'Microsoft.AspNetCore.OData, Version=8.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'.
Unknown location

TypeLoadException: Could not load type 'Microsoft.AspNet.OData.Query.HandleNullPropagationOption' from assembly 'Microsoft.AspNetCore.OData, Version=8.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'.
Unknown location

TypeLoadException: Could not load type 'Microsoft.AspNet.OData.Query.HandleNullPropagationOption' from assembly 'Microsoft.AspNetCore.OData, Version=8.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'.
Unknown location

TypeLoadException: Could not load type 'Microsoft.AspNet.OData.Query.HandleNullPropagationOption' from assembly 'Microsoft.AspNetCore.OData, Version=8.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'.
Unknown location

ReflectionTypeLoadException: Unable to load one or more of the requested types.
Could not load type 'Microsoft.AspNet.OData.Query.HandleNullPropagationOption' from assembly 'Microsoft.AspNetCore.OData, Version=8.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'.
Could not load type 'Microsoft.AspNet.OData.Query.HandleNullPropagationOption' from assembly 'Microsoft.AspNetCore.OData, Version=8.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'.
Could not load type 'Microsoft.AspNet.OData.Query.HandleNullPropagationOption' from assembly 'Microsoft.AspNetCore.OData, Version=8.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'.
Could not load type 'Microsoft.AspNet.OData.Query.HandleNullPropagationOption' from assembly 'Microsoft.AspNetCore.OData, Version=8.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'.


System.Reflection.RuntimeModule.GetTypes(RuntimeModule module)
System.Reflection.RuntimeAssembly.get_DefinedTypes()
Microsoft.AspNetCore.Mvc.ApplicationParts.AssemblyPart.get_Types()
Microsoft.AspNetCore.Mvc.Controllers.ControllerFeatureProvider.PopulateFeature(IEnumerable<ApplicationPart> parts, ControllerFeature feature)
Microsoft.AspNetCore.Mvc.ApplicationParts.ApplicationPartManager.PopulateFeature<TFeature>(TFeature feature)
Microsoft.AspNetCore.Mvc.ApplicationModels.ControllerActionDescriptorProvider.GetControllerTypes()
Microsoft.AspNetCore.Mvc.ApplicationModels.ControllerActionDescriptorProvider.GetDescriptors()
Microsoft.AspNetCore.Mvc.ApplicationModels.ControllerActionDescriptorProvider.OnProvidersExecuting(ActionDescriptorProviderContext context)
Microsoft.AspNetCore.Mvc.Infrastructure.DefaultActionDescriptorCollectionProvider.UpdateCollection()
Microsoft.AspNetCore.Mvc.Infrastructure.DefaultActionDescriptorCollectionProvider.Initialize()
Microsoft.AspNetCore.Mvc.Infrastructure.DefaultActionDescriptorCollectionProvider.GetChangeToken()
Microsoft.AspNetCore.Mvc.Routing.ActionEndpointDataSourceBase+<>c__DisplayClass11_0.<Subscribe>b__0()
Microsoft.Extensions.Primitives.ChangeToken.OnChange(Func<IChangeToken> changeTokenProducer, Action changeTokenConsumer)
Microsoft.AspNetCore.Mvc.Routing.ActionEndpointDataSourceBase.Subscribe()
Microsoft.AspNetCore.Mvc.Routing.ControllerActionEndpointDataSource..ctor(ControllerActionEndpointDataSourceIdProvider dataSourceIdProvider, IActionDescriptorCollectionProvider actions, ActionEndpointFactory endpointFactory, OrderedEndpointsSequenceProvider orderSequence)
Microsoft.AspNetCore.Builder.ControllerEndpointRouteBuilderExtensions.GetOrCreateDataSource(IEndpointRouteBuilder endpoints)
Microsoft.AspNetCore.Builder.ControllerEndpointRouteBuilderExtensions.MapControllers(IEndpointRouteBuilder endpoints)
IME.CustomerManagement.Api.OData.Broker.Startup+<>c.<Configure>b__9_0(IEndpointRouteBuilder endpoints) in Startup.cs
Microsoft.AspNetCore.Builder.EndpointRoutingApplicationBuilderExtensions.UseEndpoints(IApplicationBuilder builder, Action<IEndpointRouteBuilder> configure)
IME.CustomerManagement.Api.OData.Broker.Startup.Configure(IApplicationBuilder app, IWebHostEnvironment env, ILoggerFactory loggerFactory) in Startup.cs
System.RuntimeMethodHandle.InvokeMethod(object target, object[] arguments, Signature sig, bool constructor, bool wrapExceptions)
System.Reflection.RuntimeMethodInfo.Invoke(object obj, BindingFlags invokeAttr, Binder binder, object[] parameters, CultureInfo culture)
Microsoft.AspNetCore.Hosting.ConfigureBuilder.Invoke(object instance, IApplicationBuilder builder)
Microsoft.AspNetCore.Hosting.ConfigureBuilder+<>c__DisplayClass4_0.<Build>b__0(IApplicationBuilder builder)
Microsoft.AspNetCore.Hosting.GenericWebHostBuilder+<>c__DisplayClass15_0.<UseStartup>b__1(IApplicationBuilder app)
Microsoft.AspNetCore.Mvc.Filters.MiddlewareFilterBuilderStartupFilter+<>c__DisplayClass0_0.<Configure>g__MiddlewareFilterBuilder|0(IApplicationBuilder builder)
Microsoft.AspNetCore.Server.IIS.Core.IISServerSetupFilter+<>c__DisplayClass2_0.<Configure>b__0(IApplicationBuilder app)
Microsoft.AspNetCore.Server.IIS.Core.IISServerSetupFilter+<>c__DisplayClass2_0.<Configure>b__0(IApplicationBuilder app)
Microsoft.AspNetCore.HostFilteringStartupFilter+<>c__DisplayClass0_0.<Configure>b__0(IApplicationBuilder app)
Microsoft.AspNetCore.Hosting.GenericWebHostService.StartAsync(CancellationToken cancellationToken)
```

**Comment by xuzhg** [comment]
@kooshan Hi, I am confused that `HandleNullPropagationOption` in 8.0 preview is under namespace, `Microsoft.AspNetCore.OData.Query`, but the error message shows `Microsoft.AspNet.OData.Query.HandleNullPropagationOption`

Do you still using Microsoft.AspNetCore.OData 7.x version? Can you do a clear before build? or do a Rebuild all?

**Comment by kooshan** [comment]
Hi. 
Actually, I did that before posting this.
I suspect this conflict occurred because I also use [AutoMapper/AutoMapper.Extensions.OData](https://github.com/AutoMapper/AutoMapper.Extensions.OData) which still uses 7.x version. Although I commented out all instances of its usage which I believe should mean the 7.x dll should not be called or loaded at all, the compiler chose that version over the preview version.
My bad.
